signer/src/stacks/contracts.rs

```diff
@@ -987,11 +1070,118 @@ impl AsContractCall for RejectWithdrawalV1 {
     ///    an event on the canonical Stacks blockchain.
     /// 2. That the signer bitmap matches the signer decisions stored in
     ///    this signer's database.
-    async fn validate<C>(&self, _ctx: &C, _req_ctx: &ReqContext) -> Result<(), Error>
+    async fn validate<C>(&self, ctx: &C, req_ctx: &ReqContext) -> Result<(), Error>
     where
         C: Context + Send + Sync,
     {
-        // TODO(255): Add validation implementation
+        if self.deployer != req_ctx.deployer {
+            return Err(WithdrawalRejectErrorMsg::DeployerMismatch.into_error(req_ctx, self));
+        }
+
+        // 1. The request exists. Check whether the associated withdrawal request transaction
+        //    is confirmed on the canonical stacks blockchain. Fail the withdrawal request if
+        //    it is not on the canonical stacks blockchain.
+        //
+        // 2. The double spend check. Check whether the qualified request ID is in the
+        //    bitcoin_withdrawals_outputs table, and that any of the associated txids are
+        //    confirmed on the canonical bitcoin blockchain. Fail the withdrawal request if
+        //    such a transaction was found.
+
+        let stacks_chain_tip = ctx
+            .get_storage()
+            .get_stacks_chain_tip(&req_ctx.chain_tip.block_hash)
+            .await
+            .unwrap()
+            .unwrap();
+        let Some(report) = ctx
+            .get_storage()
+            .get_withdrawal_request_report(
+                &req_ctx.chain_tip.block_hash,
+                &stacks_chain_tip.block_hash,
+                &self.id,
+                &ctx.config().signer.public_key(),
+            )
+            .await
+            .map_err(|_| WithdrawalRejectErrorMsg::RequestMissing.into_error(req_ctx, self))?
```

Comment:
An error here should just mean that we're bailing.
```suggestion
            .await?
```

Reply:
Done